state/protocol/protocol_state/state/mutable_protocol_state_test.go

```diff
@@ -0,0 +1,650 @@
+	s.parentState.On("GetVersionUpgrade").Return(nil).Once()
+	s.parentState.On("Replicate", s.latestProtocolVersion).Return(nil, exception).Once()
+
+	// `SetupTest` initializes the mock factories to expect to be called, so we overwrite the mocks here:
+	s.kvStateMachineFactories[0] = *protocol_statemock.NewKeyValueStoreStateMachineFactory(s.T())
+	s.kvStateMachineFactories[1] = *protocol_statemock.NewKeyValueStoreStateMachineFactory(s.T())
+
+	_, dbUpdates, err := s.mutableState.EvolveState(s.candidate.ParentID, s.candidate.View, []*flow.Seal{})
+	require.Error(s.T(), err)
+	require.True(s.T(), dbUpdates.IsEmpty())
+}
+
+// Test_StateMachineFactoryFails verifies that errors received while creating the sub-state machines are escalated to the caller.
+// Protocol Convention:
+//   - The Orthogonal Store State Machines have a 3-step process to evolve their respective sub-states:
+//     (i) construction (via the injected factories)
+//     (ii) processing ordered service events from sealed results (in `EvolveState` call)
+//     (iii) in the `Build` step, the state machine assembles their resulting sub-state and the corresponding database operations
+//     to index and persist its substate.
+//   - The protocol convention is that `MutableProtocolState` executes first step (i) on all state machines, then step (ii) and lastly (iii)
+//
+// This test also verifies that the `MutableProtocolState` does not engage in step (ii) or (iii) before the completing step (i) on all state machines.
+func (s *StateMutatorSuite) Test_StateMachineFactoryFails() {
+	workingFactory := *protocol_statemock.NewKeyValueStoreStateMachineFactory(s.T())
+	stateMachine := protocol_statemock.NewOrthogonalStoreStateMachine[protocol.KVStoreReader](s.T()) // we expect no methods to be called on this state machine
+	workingFactory.On("Create", s.candidate.View, s.candidate.ParentID, &s.parentState, &s.evolvingState).Return(stateMachine, nil).Maybe()
+
+	exception := errors.New("exception")
+	failingFactory := *protocol_statemock.NewKeyValueStoreStateMachineFactory(s.T())
+	failingFactory.On("Create", s.candidate.View, s.candidate.ParentID, &s.parentState, &s.evolvingState).Return(nil, exception).Once()
+
+	s.Run("failing factory is last", func() {
+		s.kvStateMachineFactories[0], s.kvStateMachineFactories[1] = workingFactory, failingFactory //nolint:govet
+		_, dbUpdates, err := s.mutableState.EvolveState(s.candidate.ParentID, s.candidate.View, []*flow.Seal{})
+		require.Error(s.T(), err)
+		require.True(s.T(), dbUpdates.IsEmpty())
+	})
+
+	failingFactory.On("Create", s.candidate.View, s.candidate.ParentID, &s.parentState, &s.evolvingState).Return(nil, exception).Once()
+	s.Run("failing factory is first", func() {
+		s.kvStateMachineFactories[0], s.kvStateMachineFactories[1] = failingFactory, workingFactory //nolint:govet
+		_, dbUpdates, err := s.mutableState.EvolveState(s.candidate.ParentID, s.candidate.View, []*flow.Seal{})
+		require.Error(s.T(), err)
```

Comment:
```suggestion
		require.ErrorIs(s.T(), err, exception)
```

Reply:
done in https://github.com/onflow/flow-go/pull/5644/commits/9c38cfa2bb5b0d4b2e0b2dc4dd6e78c548ff2cdb